reporters/drop-ship.es

```diff
@@ -81,14 +85,18 @@ export default class DropShipReporter extends BaseReporter {
       drop.baseExp = body.api_get_base_exp
       drop.shipId = (body.api_get_ship || {}).api_ship_id || -1
       drop.itemId = (body.api_get_useitem || {}).api_useitem_id || -1
+      drop.shipCounts = drop.shipId !== -1 ? countOwnedShips(drop.shipId) : []
       drop.teitokuLv = _teitokuLv
-      this.report('/api/report/v2/drop_ship', drop)
+      drop.teitokuId = teitokuId
+      // Report enemy pattern and drops
+      this.report('/api/report/v2/drop_ship', drop).then(() => {
```

Comment:
If after `battleresult` `next` is missed (for whatever reason) next `battleresult` submission will be corrupted. There are actually reports like that in dumps, they have info from `battle` and `battleresult` from one node, but stuff from `next` is from previous node.

Reply:
Having those fileds `null`ed will at least help filtering those reports (otherwise they look normal and only can be detected by checking node id / names correspondence, since id is from `next` while name is from `battleresult`, unsynced)

Reply:
Actually, will throw (catched in base), so there will be no corrupted submission at all.